### ISSUE TITLE
Cutting down "complaining about solution one doesn't like" behaviour on solution comments?

### ISSUE THREAD

**Issue by Voileexperiments** [Codewars/codewars.com#1390]
When looking at the CW dashboard It's very common to see people ~~bitching~~complaining about top solutions they don't like in solution comments, like, once every few days? These people don't even try to make sense when complaining, like they typically deploy these common strawmans:

- One Line Is Bad (TM)
- Unreadable and Unmaintainable According To Me (TM)
- Premature Optimization (TM) (aka Use Quotes Or Arbitrary Rules I Don't Know What They Actually Mean To Bash Solution (TM))
- That's Not How You Write In Production (TM)
- Don't Get Me Wrong, I'm Into Good Code, But Blablabla (TM)

without actually addressing *why* the specific solution is bad. Very often they just pull out a strawman, continue their own multi-page rant without even mentioning anything about the solution. And they almost always says "this is clever" after "this is not best practice", as if "clever" actually means "your solution sucks".

Some users, including me, tried to argue with a few of them. Obviously they're too dumb to realize how dumb their idea is, so it didn't work at all.

This is *very* bad because they actually gain lots of traction among newbies who haven't been used to reading clean, concise code (hence you only see these in white/yellow katas). e.g: [Someone pulled this off in the top solution of Find the Odd Int](https://www.codewars.com/kata/reviews/56257b1427e918467e000174/groups/562b3fdcd9377354db00002b#59a5d919589d2a3eca000106) and it has already received *38* votes (and still growing). What I take from this is at least 38 newbies took his nonsense as something that actually make sense, and the thought of this is horrifying.

Aren't solution comments for *informing* people about the possible advantage/disadvantages of a solution? Are we supposed to look at multiple solutions and learn different approaches to a problem? Why are people instead putting unsubstantiated, misleading rants that will almost certainly put other less skilled coders to astray? I take this as a failure of CW as a learning platform. Just like draconian teachers enforcing outrageous code standard and flow, these people are no better than them.

Can we, like, put this in Code of Conduct or somewhere to stop these behaviours?

**Comment by anter69** [comment]
You can't control comments. You could simply remove the option to upvote/downvote comments, which may make the comment section "calmer"

**Comment by Voileexperiments** [comment]
Yes, but the point is, these comments are *detrimental* to CW's purpose as a learning platform, and they're the worse ones among all of these things because they actually gain traction. If someone makes a crap kata with crap tests everyone who attempts the kata is annoyed, so nobody will think that "writing crap programming problems for newbies to solve" is something that is *good*/should be done. Those "complaining about solutions one doesn't like" comments, however, enforces the common misbelieves and myth. They mislead less aware people into thinking they're universal rules that should be obeyed, like that draconian college teacher who insists that you should not return early because reasons (being a stylish problem aside, it's also *wrong* in most cases). And there lacks people objecting to this notion directly.

I mean, yes, my expectations for an average programmer is pretty something after a year of CW experience, but I still believe that *feeding newbies subjective belief and mostly wrong information* is wrong and should be stopped, especially considering that CW is a learning platform. And bitching about solutions one doesn't like is like the most rock-star way of doing that.

**Comment by anter69** [comment]
That's fine, but how do you identify such comments? By an algo? I doubt it... By employing moderators? Not on CW. You spot it by chance? Yes. Would you miss another 1000 similar comments? Surely.

There could be "report" button for comments, yes. But who would review those reports?

**Comment by jhoffner** [comment]
> Yes, but the point is, these comments are detrimental to CW's purpose as a learning platform, and they're the worse ones among all of these things because they actually gain traction.

From the example you gave and having just read through it, other than some frustration starting to bubble up which made it get slightly more emotional than it really needed to be, I don't personally see anything wrong with this. So the OP has a subjective issue with this type of code, and then a subjective conversation broke out. Maybe someone learned something. Maybe a new perspective was obtained by someone writing in or at least reading the thread. People have different opinions, if we can keep things civil then we can all learn from each other. 

Honestly this thread itself is basically complaining about complaining. I would love to make it more productive and talk about solutions. 

For me, encouraging more comments within solutions, especially in cases like this where there is something that is arguably clever and best practice - I think would go a long way in providing context to the "complainers" as well as be much more educational for the site as a whole. 

Maybe there could be a separate "Well documented" voting button which is more like a karma vote, not an up-vote. However that could get complicated when it comes to figuring out which solution within a solution group to show.

**Comment by nomennescio** [comment]
"Beauty is in the eye of the beholder"

The whole "social" part of CW is largely driven by people's opinions. You cannot put an objective measure to that. I know that in my code I do not shy away from breaking some commonly hold conventions, because I think they are wrong. That will never make such code popular.

**Comment by Voileexperiments** [comment]
> The whole "social" part of CW is largely driven by people's opinions. You cannot put an objective measure to that.

Sure, but also note that the majority of users in CW are newbies, so by that you're also implying you have no problems to have CW largely driven by newbies, who'd have absolutely no problems in doing things like:
- thinking imperative code is the most readable code
- misusing the phrase "clever"
- posting hot garbage beta katas
- upvoting bad/duplicate beta katas happily
- raising bogus issues for bugs in their code

and lots more. And it's not the first day that the wordings in some parts of CW is just *bad*: satisfaction rating is still punitive, and the distinction between issue and question is still not clear at all; the wording "clever" is just but one of them, and it's a nasty one.

It's easy to just say "CW is community-driven so let's just leave the people to do what they want", but don't forget that it doesn't mean there are no *responsibilities*. CW still has the responsibility to keep up its quality as a competitive coding and learning platform, and that's part of community management.

**Comment by nomennescio** [comment]
Others have made the point that we don't know who CW newbies are and what their skills are. And I've seen pretty uninformed people which high honor count. I'm not against any system that introduces some kind of weight to the vote of people with high *ranks* based on language. It will not be perfect either.
